jellyfin-api/src/main/kotlin/org/jellyfin/sdk/api/client/KtorClient.kt

```diff
@@ -207,22 +214,39 @@ public open class KtorClient(
 		queryParameters: Map<String, Any?> = emptyMap(),
 		requestBody: Any? = null,
 	): Response<T> {
+		val logger = LoggerFactory.getLogger(this::class.java)
 		val url = createUrl(pathTemplate, pathParameters, queryParameters)
 
 		// Log HTTP call with access token removed
 		val safeUrl = accessToken?.let { url.replace(it, "******") } ?: url
-		LoggerFactory.getLogger(this::class.java).info("$method $safeUrl")
+		logger.info("$method $safeUrl")
 
-		val response = client.request<HttpResponse> {
-			this.method = method
-			url(url)
-			header(HttpHeaders.Authorization, createAuthorizationHeader())
+		@Suppress("SwallowedException")
+		try {
+			val response = client.request<HttpResponse> {
+				this.method = method
+				url(url)
+				header(HttpHeaders.Authorization, createAuthorizationHeader())
 
-			if (requestBody != null)
-				body = defaultSerializer().write(requestBody)
-		}
+				if (requestBody != null) body = defaultSerializer().write(requestBody)
+			}
 
-		return Response(response.receive(), response.status.value, response.headers.toMap())
+			// Check HTTP status
+			if (!response.status.isSuccess()) throw InvalidStatusError(response.status.value)
+			// Read response body
+			val body = response.receive<T>()
+			// Return custom response instance
+			return Response(body, response.status.value, response.headers.toMap())
+		} catch (err: HttpRequestTimeoutException) {
+			logger.debug("Connection timed out", err)
+			throw TimeoutError(err.message)
+		} catch (err: NoTransformationFoundException) {
+			logger.error("Requested model does not exist!?", err)
```

Comment:
`!?` is a bit aggressive, but sure :sweat_smile:

Reply:
If we used the Android logger this would be a `logger.wtf` because this should not be possible to happen.

Reply:
Unrelated, but I still love that method name. *What a terrible failure* - yeah sure.